lib/diskoverflow/sorted.go

```diff
@@ -0,0 +1,384 @@
+// Copyright (C) 2018 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package diskoverflow
+
+import (
+	"bytes"
+	"sort"
+)
+
+// Sorted stores a list of SortValue sorted by the return value of their Key method.
+// After calling PopFirst or PopLast for the first time, calls to Add will panic.
+type Sorted interface {
+	Common
+	Add(v SortValue)
+	Iter(fn func(v Value) bool, rev bool)
+	IterAndClose(fn func(v Value) bool, rev bool)
+	PopFirst() (Value, bool)
+	PopLast() (Value, bool)
+}
+
+func NewSorted(location string) Sorted {
+	s := &sorted{
+		key:      lim.register(),
+		location: location,
+	}
+	s.commonSorted = &memorySorted{key: s.key}
+	return s
+}
+
+// SortValue must be implemented by every supported type for sorting. The sorting
+// will happen according to bytes.Compare on the key.
+type SortValue interface {
```

Comment:
Why can't this just be
```
type SortValue interface {
  LessThan(other SortValue)
  Value
}
```

Reply:
I need a key to store it in the key-value-database.